### PR TITLE
adding and editing docs for cluster role aggregation

### PR DESCRIPTION
**Pull request by kalexand-rh**
Continuing work on https://github.com/openshift/openshift-docs/pull/7363

### CODE REVIEW

**Comment by kalexand-rh** [comment]
@enj, I modified the information in the architecture guide and added steps to the topic about creating CRDs. PTAL.

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: I assume this was dropped in favor of pointing to the Kube docs?

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: definition

File: architecture/additional_concepts/authorization.adoc
Comment (enj) [inline architecture/additional_concepts/authorization.adoc]: We should probably link to the Kube docs so there is at least some example of how to create a new cluster role that gets aggregated into (as apposed to cluster roles that aggregate into other cluster roles).

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: Wording here is a bit weird.

If you want to grant access to the CRD to normal users, the simplest way is to leverage cluster role aggregation via the standard cluster roles admin, edit, and view.  admin or edit is generally assigned to the user that created a project via the project template.  Cluster role aggregation allows the insertion of custom policy rules into these cluster roles.  This seamlessly integrates the new resource into the cluster's RBAC policy as if it was a built-in resource.

**Comment by enj** [comment]
@kalexand-rh how do you tell the bot to regenerate the docs?  I just did an initial pass so far.

File: architecture/additional_concepts/authorization.adoc
Comment (kalexand-rh) [reply architecture/additional_concepts/authorization.adoc]: I didn't know you could create a cluster role. If that's true, the steps for creating a cluster role should go in https://docs.openshift.com/container-platform/3.9/admin_guide/manage_rbac.html, not here, and describe the option to create a role that uses aggregation.
Can we talk about this one?

File: admin_guide/custom_resource_definitions.adoc
Comment (kalexand-rh) [inline admin_guide/custom_resource_definitions.adoc]: I think "The admin or edit role is usually assigned to the user that created a project through the project template. " goes at the end of this note, not in the intro.

File: admin_guide/custom_resource_definitions.adoc
Comment (kalexand-rh) [reply admin_guide/custom_resource_definitions.adoc]: Thanks! I think the entire change should be replaced with this text:

or a single project (namespace). As with existing built-in objects, deleting a project deletes all custom objects in that project.

If you want to grant access to the CRD to users, use cluster role aggregation to grant access to users with the admin, edit, or view default cluster roles. Cluster role aggregation allows the insertion of custom policy rules into these cluster roles. This behavior seamlessly integrates the new resource into the cluster's RBAC policy as if it was a built-in resource.

**Comment by kalexand-rh** [comment]
@enj, this is the only place I've ever seen that bot, and I think you have to update the PR to regenerate them. I'll incorporate your comments now and update the PR.

File: admin_guide/custom_resource_definitions.adoc
Comment (kalexand-rh) [reply admin_guide/custom_resource_definitions.adoc]: Yup, on line 198.

**Comment by kalexand-rh** [comment]
@xltian, @openshift/team-documentation  PTAL

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: The roles with

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: the "get crontabs" permission

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: I think this should be at the end since it is assumed that cluster admin is performing these actions and thus can perform all actions.

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: Should says cluster role?

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: name: aggregate-cron-tabs-admin-edit

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: Specify this label to grant

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: Same

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: Should mention read and write permissions.

File: admin_guide/custom_resource_definitions.adoc
Comment (enj) [inline admin_guide/custom_resource_definitions.adoc]: Should mention read only.

**Review by enj — CHANGES_REQUESTED** [review]

File: admin_guide/custom_resource_definitions.adoc
Comment (kalexand-rh) [reply admin_guide/custom_resource_definitions.adoc]: I'm not sure which change you'd like. Will you please elaborate?

File: admin_guide/custom_resource_definitions.adoc
Comment (kalexand-rh) [reply admin_guide/custom_resource_definitions.adoc]: @enj, is this better?

File: admin_guide/custom_resource_definitions.adoc
Comment (kalexand-rh) [reply admin_guide/custom_resource_definitions.adoc]: Ah! Thanks! :)

**Comment by kalexand-rh** [comment]
@enj, here's a rendering of the admin and dev versions. I think the intro in the dev version should be fine. Please scan and let me know if you see issues. 
admin version: http://file.rdu.redhat.com/kalexand/051018/pr7363/admin_guide/custom_resource_definitions.html
dev version: http://file.rdu.redhat.com/kalexand/051018/pr7363/dev_guide/creating_crd_objects.html

**Comment by kalexand-rh** [comment]
@bfallonf, PTAL at the logic for the modularization for the two topics. :)

**Comment by bfallonf** [comment]
@kalexand-rh Looks all correct to me. Up to you if you want to give a quick edit of the existing content that's there while you're playing with these files. There's a few nits I would've made :truck:

**Comment by xltian** [comment]
@stuartchuan  Please review on behalf of QE, let me know if this is not owned by your team, thanks.

**Comment by stuartchuan** [comment]
@kalexand-rh Looks all changes good to me.

**Comment by kalexand-rh** [comment]
@bfallonf, I'd love to hear about your nits!

File: admin_guide/topics/crd_overview.adoc
Comment (enj) [inline admin_guide/topics/crd_overview.adoc]: can create an instance of a CRD

File: admin_guide/topics/create_aggregated_cluster_role_crd.adoc
Comment (enj) [inline admin_guide/topics/create_aggregated_cluster_role_crd.adoc]: link to https://docs.openshift.org/latest/admin_guide/managing_projects.html#template-for-new-projects

File: admin_guide/topics/create_aggregated_cluster_role_crd.adoc
Comment (enj) [inline admin_guide/topics/create_aggregated_cluster_role_crd.adoc]: Missing deletecollection

`["get", "list", "watch", "create", "update", "patch", "delete", "deletecollection"]`

File: dev_guide/creating_crd_objects.adoc
Comment (enj) [inline dev_guide/creating_crd_objects.adoc]: Dev guide probably needs a snippet that says "let us assume the cluster admin has set up the following CRD for you in advance and has given you read and write access to it."

**Comment by enj** [comment]
@kalexand-rh very minor nits, feel free to ignore if you disagree.

File: admin_guide/topics/crd_overview.adoc
Comment (kalexand-rh) [reply admin_guide/topics/crd_overview.adoc]: @enj, so is it ever right to say that you can create an object from a CRD? That language shows up in a couple of other spots.

Comment (enj) [reply admin_guide/topics/crd_overview.adoc]: Based on https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/#create-custom-objects I think it is fine as-is.

**Comment by kalexand-rh** [comment]
@openshift/team-documentation, PTAL

File: admin_guide/topics/crd_overview.adoc
Comment (bmcelvee) [inline admin_guide/topics/crd_overview.adoc]: s/behavior seamlessly integrates/behavior integrates

File: admin_guide/topics/manage_crd_obj.adoc
Comment (bmcelvee) [inline admin_guide/topics/manage_crd_obj.adoc]: The <1> & <2> aren't rendering correctly in the build, though they do the job of calling out the line items.

**Comment by bmcelvee** [comment]
Couple of tiny nits + rebase needed, otherwise LGTM! :cat2:

**Comment by kalexand-rh** [comment]
Thank you, @bmcelvee! I fixed the nits, rebased, and get to wait for Jenkins. :)

**Pull request merged**
